sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -70,21 +108,44 @@ export class EventHubSender extends LinkEntity {
    * The AMQP sender link.
    */
   private _sender?: AwaitableSender;
+  /**
+   * Indicates whether the sender is configured for idempotent publishing.
+   */
+  private _isIdempotentProducer: boolean;
+  /**
+   * Indicates whether the sender has an in-flight send while idempotent
+   * publishing is enabled.
+   */
+  private _hasPendingSend?: boolean;
+  /**
+   * A local copy of the PartitionPublishingProperties that can be mutated to
+   * keep track of the lastSequenceNumber used.
+   */
+  private _localPublishingProperties?: PartitionPublishingProperties;
+  /**
+   * The user-provided set of options that can be specified to influence
+   * publishing behavior specific to a partition.
+   */
+  private _userProvidedPublishingOptions?: PartitionPublishingOptions;
 
   /**
    * Creates a new EventHubSender instance.
    * @hidden
    * @param context - The connection context.
-   * @param partitionId - The EventHub partition id to which the sender
-   * wants to send the event data.
+   * @param options - Options used to configure the EventHubSender.
    */
-  constructor(context: ConnectionContext, partitionId?: string) {
+  constructor(
+    context: ConnectionContext,
+    { partitionId, enableIdempotentProducer, partitionPublishingOptions }: EventHubSenderOptions
```

Comment:
The main reason to blow them out here is because then I can reference `partitionId` in the `super()` call. TypeScript complains if you try to do something like `const partitionId = options.partitionId` before the super call.

It's not a big deal, I could update the super call to use options.partitionId instead of partitionId, but that's why I did it.